Update sidebar scrolling

tlesicka opened the pull request:
Splitting #3457 into several PR's as requested by @MatissJanis.
Refactored sidebar to allow scrolling in only the accounts section.
Moved Item buttons to PrimaryButtons.
Put Tools inside PrimaryButtons and removed Tools.
Moved Add Account button to SecondaryButtons. Moved account sorting from sidebar to Accounts.

tlesicka commented on packages/desktop-client/src/components/sidebar/Accounts.tsx:
```diff
@@ -61,108 +57,129 @@ export function Accounts({
+            connected={!!account.bank}
+            pending={syncingAccountIds.includes(account.id)}
+            failed={failedAccounts && failedAccounts.has(account.id)}
+            updated={updatedAccounts && updatedAccounts.includes(account.id)}
```
Applied this change

tlesicka commented on packages/desktop-client/src/components/sidebar/Accounts.tsx:
```diff
@@ -61,108 +57,129 @@
     return null;
   };
 
+  async function onReorder(
+    id: string,
+    dropPos: 'top' | 'bottom',
+    targetId: unknown,
+  ) {
```
This code was removed from Sidebar.tsx and placed in Accounts.tsx.  I will leave this as is.

tlesicka commented on packages/desktop-client/src/components/sidebar/Accounts.tsx:
```diff
@@ -61,108 +57,129 @@ export function Accounts({
     return null;
   };
 
+  async function onReorder(
+    id: string,
+    dropPos: 'top' | 'bottom',
+    targetId: unknown,
+  ) {
+    let targetIdToMove = targetId;
+    if (dropPos === 'bottom') {
+      const idx = accounts.findIndex(a => a.id === targetId) + 1;
+      targetIdToMove = idx < accounts.length ? accounts[idx].id : null;
+    }
+
+    dispatch(moveAccount(id, targetIdToMove));
+  }
```
Leaving this code as is because it was only moved from Sidebar.tsx to Accounts.tsx

MatissJanis commented on packages/desktop-client/src/components/sidebar/SecondaryButtons.tsx:
```diff
@@ -0,0 +1,38 @@
+// @ts-strict-ignore
```
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏new components we create ideally should be strict TS compliant.

```suggestion
```

tlesicka replied:
Removed

MatissJanis commented on packages/desktop-client/src/components/sidebar/PrimaryButtons.tsx:
```diff
@@ -1,3 +1,4 @@
+// @ts-strict-ignore
```
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what problem do we have here that this is added?
```suggestion
```

tlesicka replied:
Removed

MatissJanis commented on packages/desktop-client/src/components/sidebar/Accounts.tsx:
```diff
@@ -43,15 +37,17 @@ export function Accounts({
   );
 
-  const getAccountPath = account => `/accounts/${account.id}`;
+  const getAccountPath = (account: AccountEntity) => `/accounts/${account.id}`;
```
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏thanks!

tlesicka replied:
Thanks, I "typed" the last couple ts errors and removed @ts-strict-ignore from this file.

MatissJanis reviewed (changes requested):
Overall LGTM. But noticed two issues:

1. the close sidenav icon is no longer in the correct location
2. there should be a min-height of some sort for the scrollable account region; on smaller screen resolutions this block is entirely invisible 

<img width="374" alt="Screenshot 2024-10-05 at 18 27 13" src="https://github.com/user-attachments/assets/fa140122-ac4d-4ee8-b5f6-af5b0e5e4b4d">
<img width="499" alt="Screenshot 2024-10-05 at 18 26 50" src="https://github.com/user-attachments/assets/ac6448c4-bbac-4e42-ae7a-dd1c696abcfd">

tlesicka commented on packages/desktop-client/src/components/sidebar/Accounts.tsx:
```diff
@@ -43,15 +36,17 @@
+    'ui.showClosedAccounts',
+  );
 
   function onDragChange(drag) {
```
Updated "types" as indicated and removed ```@ts-strict-ignore```

tlesicka commented on packages/desktop-client/src/components/sidebar/Accounts.tsx:
```diff
@@ -43,126 +39,149 @@ export function Accounts({
+              onDragChange={onDragChange}
+              onDrop={onReorder}
+            />
+          ))}
```
Updated as indicated

tlesicka commented:
@MatissJanis  Thanks for the previous review.  Fixed the errors as indicated below.

> 1) the close sidenav icon is no longer in the correct location

In copy/paste from other PR, a ```<View>``` was deleted.  Found the error and fixed it.

> 2) there should be a min-height of some sort for the scrollable account region; on smaller screen resolutions this block is entirely invisible

Fixed.  Now, if the screen height drops below 480px, the scrollable area reverts back to the previous scrollable area (primary buttons, account list, secondary buttons) and the account list is full size.

tlesicka commented on packages/desktop-client/src/components/sidebar/Accounts.tsx:
```diff
@@ -43,126 +36,156 @@ export function Accounts({
+    return undefined;
+  };
+
+  async function onReorder(
+    id: string,
+    dropPos: 'top' | 'bottom',
+    targetId: unknown,
+  ) {
+    let targetIdToMove = targetId;
+    if (dropPos === 'bottom') {
+      const idx = accounts.findIndex(a => a.id === targetId) + 1;
+      targetIdToMove = idx < accounts.length ? accounts[idx].id : null;
+    }
+
+    dispatch(moveAccount(id, targetIdToMove));
+  }
```
I am not changing targetId at this time.  It will take more work than just this file and is above the scope of this PR.  Please keep reminding me of this issue and I will try to fix it in a future PR.

MatissJanis reviewed (approved):
LGTM on my end! Thanks for splitting this out into a smaller PR.

(I won't merge the PR - will wait for the folks in the other PR to also give feedback here if they have any since this is a very visible part of the product)

@jfdoming @RubenOlsen @Teprifer @Jonathan-Fang

Jonathan-Fang commented:
Thank you @tlesicka for your efforts to implement these sidebar changes, and thanks @MatissJanis for the mention to remind providing feedback!

LGTM!

Teprifer commented:
The add account took me a sec to find tbh, and I initially worried how discoverable it would be for new users but if there's no accounts Actual already gives a nice obvious notice and link to create one, so no issue there (and this is the initial page shown to new users).

![image](https://github.com/user-attachments/assets/0448f6c6-b171-4c8b-95ca-97e50364ae5e)

I think it's ok to merge @MatissJanis

matt-fidd commented:
LGTM too @MatissJanis 
VRT will need rerunning to fix the conflicts.
